airflow/providers/databricks/operators/databricks.py

```diff
@@ -872,7 +896,23 @@ def execute_complete(self, context: Context, event: dict[str, Any] | None = None
             _handle_deferrable_databricks_operator_completion(event, self.log)
             if event["repair_run"]:
                 self.repair_run = False
+                run_state = RunState.from_json(event["run_state"])
                 self.run_id = event["run_id"]
+                job_id = self._hook.get_job_id(self.run_id)
+                repair_reason = next(
+                    (
+                        reason
+                        for reason in (self.databricks_repair_reason_new_settings or {}).keys()
+                        if reason in run_state.state_message
+                    ),
+                    None,
+                )
+                if repair_reason is not None and self.databricks_repair_reason_new_settings:
+                    new_settings_json = normalise_json_content(
+                        self.databricks_repair_reason_new_settings[repair_reason]
+                    )
+                    self._hook.update_job(job_id=str(job_id), json=new_settings_json)
```

Comment:
1. Same comment regarding adding tests.
2. Looks repetitive - maybe it's worth extracting into a util function/static method and reuse it?

Reply:
+1

Reply:
sure working on it